winrm/client.go

```diff
@@ -23,21 +29,38 @@ type Client struct {
 
 // NewClient will create a new remote client on url, connecting with user and password
 // This function doesn't connect (connection happens only when CreateShell is called)
-func NewClient(endpoint *Endpoint, user, password string) (client *Client, err error) {
+func NewClient(endpoint *Endpoint, user, password string, auth AuthType) (client *Client, err error) {
```

Comment:
An alternative would be to add a new exported function like ```NewX509Client``` which accepts key and publicKey instead of user/pass and  calls ```NewClientWithParameters``` with appropriate parameters. That way you get to keep current API, and you just add a new function.

Thoughts?